state/execution.go

```diff
@@ -924,12 +942,18 @@ func (blockExec *BlockExecutor) saveFailedProposalBlock(state State, block *type
 	}
 
 	debugDir := filepath.Join(blockExec.rootDir, "data", "debug")
-	filename := fmt.Sprintf("%s-%d-%s_failed_proposal.pb",
+	baseFilename := fmt.Sprintf("%s-%d-%s_failed_proposal.pb",
 		state.ChainID,
 		block.Height,
 		reason,
 	)
 
+	filename, err := FindNextAvailableFilename(debugDir, baseFilename)
```

Comment:
remove this function. And in the base filename in line 945, just add the timestamp to the filename and keep it called `filename`